cnxpublishing/views/admin.py

```diff
@@ -271,3 +274,86 @@ def admin_edit_site_message_POST(request):
     args = admin_edit_site_message(request)
     args['response'] = "Message successfully Updated"
     return args
+
+
+@view_config(route_name='admin-print-style', request_method='GET',
+             renderer='cnxpublishing.views:templates/print-style.html',
+             permission='administer')
+def admin_print_styles(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+    styles = []
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""\
+                SELECT print_style, fileid, recipe_type, revised, tag,
+                    (SELECT count (*) from latest_modules as lm
+                        where lm.print_style=ps.print_style
+                            and lm.portal_type='Collection')
+                FROM print_style_recipes as ps;""")
+            for row in cursor.fetchall():
+                styles.append({
+                    'print_style': row[0],
+                    'file': row[1],
+                    'type': row[2],
+                    'revised': row[3],
+                    'tag': row[4],
+                    'number': row[5],
+                    'link': request.route_path('admin-print-style-single',
+                                               style=row[0])
+                })
+    return {'styles': styles}
+
+
+@view_config(route_name='admin-print-style-single', request_method='GET',
+             renderer='cnxpublishing.views:templates/print-style-single.html',
+             permission='administer')
+def admin_print_styles_single(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+    style = request.matchdict['style']
+    args = {}
+    # do db search to get file id and other info on the print_style
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""
+                SELECT print_style, fileid, recipe_type, tag
+                from print_style_recipes
+                WHERE print_style=%s
+                """, vars=(style,))
+            info = cursor.fetchall()
+            if len(info) != 1:
```

Comment:
I had a look at the [print_style_recipes table](https://github.com/Connexions/cnx-db/blob/master/cnxdb/archive-sql/schema/tables.sql#L285), `print_style` is not unique, `(print_style, tag)` is.  It is possible that this query will return more than one row.  Are we showing the latest one or do we need to show all the versions of it?  If we are just showing one, we probably want to add `ORDER BY revised DESC LIMIT 1` to the SQL statement.